crates/sui-indexer-alt-framework/src/ingestion/client.rs

```diff
@@ -102,6 +102,13 @@ impl IngestionClient {
         Ok(Self::new_impl(Arc::new(client), metrics))
     }
 
+    #[cfg(test)]
+    pub(crate) fn new_for_testing(client: Arc<dyn IngestionClientTrait>) -> Self {
```

Comment:
The things that are specific to testing in this function are to do with metrics. It would be better to introduce a `new_for_testing` on `IndexerMetrics` for this.